src/index.js

```diff
@@ -1,24 +1,125 @@
 import './style.css';
 
-const lists = [
-  { description: 'Wash the dishes', completed: false, index: 1 },
-  { description: 'Complete to-do list project', completed: false, index: 2 },
-];
-
-const provideLists = () => {
-  const placeholder = document.querySelector('.todo-placeholder');
-  placeholder.innerHTML = '';
-  const sortedLists = lists.sort((a, b) => a.index - b.index);
-  sortedLists.forEach((list) => {
-    const listItem = document.createElement('div');
-    listItem.className = 'todo';
-    listItem.innerHTML = `
-        <div>
-          <input type="checkbox" class="list" ${list.completed ? 'checked' : ''} />${list.description}
-        </div>
-        <i class="fa-solid fa-ellipsis-vertical"></i>
-      `;
-    placeholder.appendChild(listItem);
+{
+  const lists = JSON.parse(localStorage.getItem('lists')) || [];
+
+  const saveLists = () => {
+    localStorage.setItem('lists', JSON.stringify(lists));
+  };
+
+  const updateIndexes = () => {
+    lists.forEach((list, index) => {
+      list.index = index + 1;
+    });
+  };
+
+  const provideLists = () => {
+    const placeholder = document.querySelector('.todo-placeholder');
+    placeholder.innerHTML = '';
+    const sortedLists = lists.sort((a, b) => a.index - b.index);
+    sortedLists.forEach((list) => {
+      const listItem = document.createElement('div');
+      listItem.className = 'todo';
+      const checkedClass = list.completed ? 'checked' : '';
+      const textDecoration = list.completed ? 'line-through' : 'none';
+      listItem.innerHTML = `
+          <div>
+            <input type="checkbox" class="list" ${list.completed ? 'checked' : ''} data-index="${list.index}" />
+            <span class="list-description ${checkedClass}" data-index="${list.index}" style="text-decoration: ${textDecoration}">${list.description}</span>
+          </div>
+          <i class="fa-solid fa-ellipsis-vertical" data-index="${list.index}"></i>
+        `;
+      placeholder.appendChild(listItem);
+    });
+  };
+
+  const addTask = (description) => {
+    const newTask = {
+      description,
+      completed: false,
+      index: lists.length + 1,
+    };
+    lists.push(newTask);
+    provideLists();
+    saveLists();
+  };
+
+  const deleteTask = (index) => {
+    lists.splice(index, 1);
+    updateIndexes();
+    provideLists();
+    saveLists();
```

Comment:
- [ ] Please create a new JavaScript file for the add and remove to-do item functionality as it is a compulsory requirement 🙏 
![image](https://github.com/mryeminthu/To-Do-list/assets/46839548/f6ce4e2e-67db-464c-b9f9-7ebc384123e5)